Tests/AlicerceTests/Network/JSONTests.swift

```diff
@@ -1531,6 +1531,121 @@ class JSONTests: XCTestCase {
+        let testDate = Date()
+        let dateFormatter: DateFormatter = {
+            $0.dateFormat = "yyyy-MM-dd"
+            return $0
+        }(DateFormatter())
+
+        let json = [
+            "testDate" : dateFormatter.string(from: testDate)
+        ]
+
+        do {
+            let date = try JSON.parseDateAttribute(key: "testDate", json: json, formatter: dateFormatter.date)
+
+            XCTAssertEqual(testDate, date)
+        } catch {
+            XCTFail("🔥: unexpected error \(error)")
+        }
+    }
+
+    func testParseDateAttribute_WithInvalidInteger_ShouldFailWithUnexpectedAttributeValue() {
+
+        let json = [
+            "testDate" : 1531
+        ]
+
+        do {
+            let _ = try JSON.parseDateAttribute(key: "testDate",
+                                                   json: json,
+                                                   formatter: Date.init(timeIntervalSince1970:))
+
+            XCTFail("🔥: unexpected success!")
+        } catch let JSON.Error.unexpectedAttributeValue(key, json: _) {
+            XCTAssertEqual(key, "testDate")
+        } catch {
+            XCTFail("🔥: unexpected error \(error)")
+        }
+    }
+
+    func testParseDateAttribute_WithInvalidStringForFormatter_ShouldFailWithUnexpectedAttributeValue() {
+
+        let dateFormatter: DateFormatter = {
+            $0.dateFormat = "yyyy-MM-dd"
+            return $0
+        }(DateFormatter())
+
+        let json = [
+            "testDate" : "🤷‍♂️"
+        ]
+
+        do {
+            let _ = try JSON.parseDateAttribute(key: "testDate", json: json, formatter: dateFormatter.date)
+
+            XCTFail("🔥: unexpected success!")
+        } catch let JSON.Error.unexpectedAttributeValue(key, json: _) {
+            XCTAssertEqual(key, "testDate")
+        } catch {
+            XCTFail("🔥: unexpected error \(error)")
+        }
+    }
+
+    func testParseOptionalDateAttribute_WithUnexistingKey_ShouldReturnNil() {
+
+        let json = [
+            "ups" : "🤷‍♂️"
+        ]
+
+        do {
+            let date = try JSON.parseOptionalDateAttribute(key: "testDate",
+                                                           json: json,
+                                                           formatter: Date.init(timeIntervalSince1970:))
+
+            XCTAssertNil(date)
+        } catch {
+            XCTFail("🔥: unexpected error \(error)")
+        }
+    }
+
+    func testParseOptionalDateAttribute_WithInvalidIntegerInferredFromFormatter_ShouldReturnNil() {
+
+        let json = [
+            "testDate" : 1513
+        ]
+
+        do {
+            let date = try JSON.parseOptionalDateAttribute(key: "testDate",
+                                                           json: json,
+                                                           formatter: Date.init(timeIntervalSince1970:))
+
+            XCTAssertNil(date)
+        } catch {
+            XCTFail("🔥: unexpected error \(error)")
+        }
+    }
+
```

Comment:
There's a missing test where the raw value is right, but the conversion fails with an error (which currently returns `nil`)

Reply:
this one 👉 `testParseDateAttribute_WithInvalidStringForFormatter_ShouldFailWithUnexpectedAttributeValue`

Reply:
Yes, I think so 😉
Also, we should do the same for the `parseOptionalDateAttribute` variant 😇

Reply:
Yes, it is 😉 
Still, we should do the same test for the `parseOptionalDateAttribute` variant 😇